.github/workflows/test-build-deploy.yml

```diff
@@ -359,10 +376,24 @@ jobs:
           echo "Running integration tests (group ${{ matrix.test_group_id }} of ${{ matrix.test_group_total }}) with Go version: $(go version)"
           ./.github/workflows/scripts/run-integration-tests-group.sh --index ${{ matrix.test_group_id }} --total ${{ matrix.test_group_total }}
 
+  checkDeploy: 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Whether In Deploy Process
+        id: check_deploy
+        run: |
+          if [[ "${{ startsWith(github.ref, 'refs/tags/') || startsWith(github.ref, 'refs/heads/r') }}" == 'true' && "${{ github.event_name == 'push' }}" == 'true' && "${{ github.repository == 'grafana/mimir' }}" == 'true' ]]; then
+            echo "is_deploy=true" >> $GITHUB_OUTPUT
+          else
+            echo "is_deploy=false" >> $GITHUB_OUTPUT
+          fi
```

Comment:
can we please use either shell script or github expression, but not both at the same time? It's very confusing.

Reply:
changed to joshua's idea, output directly with github expression